mapproxy/util/ext/serving.py

```diff
@@ -119,8 +123,24 @@ def run_wsgi(self):
 
         def write(data):
             assert headers_set, 'write() before start_response'
+            import pdb; pdb.set_trace()
             if not headers_sent:
-                status, response_headers = headers_sent[:] = headers_set
+                if self.request_version == "HTTP/1.0":
+                    headers_sent[:] = headers_set
+                    status = headers_set[0]
+                    # remove non-HTTP1.0 headers
+                    response_headers = (
```

Comment:
i cannot see any removal here, i think you are doing the opposite and keep all unwanted headers

Reply:
OK, I'll update the comment. The list contains all http 1.0 headers I found, so the logic is correct, but is not in line with the comment